File: dwite08c3p4.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
  int m, n, t;
  for (int x = 0; x < 5; x++){
    vector<int> k(101, 100);
    k[1] = 0;
    vector<int> currencies;
    cin >> m;
    cin >> n;

    for (int y = 0; y<n; y++){
      cin >> t;
      currencies.push_back(t);
    }

    for (int i = 1; i<k.size(); i++){
      if (k[i] == 100){
        continue;
      }
      for (int l : currencies){
        if (i+l < k.size()){
          k[i+l] = min(k[i]+1, k[i+l]);
        }
      }
    }
    cout << k[m+1] << endl;
  }

  return 0;
}

File: dwite08c3p4.py
for x in range(5):
    k = [float('inf') for i in range(101)]
    k[1] = 0
    currencies = []
    m = int(input())
    n = int(input())

    for y in range(n):
        currencies.append(int(input()))
    
    for i in range(1, len(k)):
        if k[i] == float('inf'):
            continue
        for l in currencies:
            if i+l < len(k):
                k[i+l] = min(k[i]+1, k[i+l])
        
    print(k[m+1])